sunpy/net/helioviewer.py

```diff
@@ -23,44 +23,86 @@
 
 class HelioviewerClient(object):
     """Helioviewer.org Client"""
-    def __init__(self, url="https://legacy.helioviewer.org/api/"):
+    def __init__(self, url="https://api.helioviewer.org/"):
         """
         url : location of the Helioviewer API.  The default location points to
-            version 1 of the API.  Version 1 of the Helioviewer API is
-            currently planned to be supported until the end of April 2017.
+            version 2 of the API.
         """
         self._api = url
 
     def get_data_sources(self, **kwargs):
         """
         Returns a structured list of datasources available at helioviewer.org.
+
+        Parameters
+        ----------
+        verbose: `bool`
+            (Optional) If set to True, it outputs the hierarchical list in an
+            alternative format and limits the list of available datasources
+            to a known set (SDO and SOHO).
+        enable: `str`
+            (Optional) Comma-separated list of observatories to enable.
+        callback: `str`
+            (Optional) Wrap the response object in a function call of your choosing.
+
+        Returns
+        -------
+        out : `dict`
+            A dictionary containing meta-information for each data source
+            that Helioviewer supports.
         """
         params = {"action": "getDataSources"}
-        params.update(kwargs)
-
+        params.update(**kwargs)
         return self._get_json(params)
-
-    def get_closest_image(self, date, **kwargs):
+    
+    def get_source_id(self):
+        """
+        Returns a dictionary containing the source ID of all the
+        images in the data source
+        """
+        d = dict()
+        from sunpy.net.helioviewer import HelioviewerClient
+        hv = HelioviewerClient()  # doctest: +REMOTE_DATA
+        datasources = hv.get_data_sources() # doctest: +REMOTE_DATA
+        for name, observatory in datasources.items():  # doctest: +REMOTE_DATA
+            if name == "TRACE":  # doctest: +REMOTE_DATA
+                for instrument, params in observatory.items():  # doctest: +REMOTE_DATA
+                    d[(name, instrument)] = params['sourceId']  # doctest: +REMOTE_DATA
+            else:  # doctest: +REMOTE_DATA
+                for inst, detectors in observatory.items():  # doctest: +REMOTE_DATA
+                    for wavelength, params in detectors.items():  # doctest: +REMOTE_DATA
+                        if name in ["Hinode", "STEREO_A", "STEREO_B"] or wavelength in ["C2", "C3"]:  # doctest: +REMOTE_DATA
+                            for wave, adict in params.items():  # doctest: +REMOTE_DATA
+                                d[(name, inst, wavelength, wave)] = adict['sourceId']  # doctest: +REMOTE_DATA
+                        else:  # doctest: +REMOTE_DATA
+                            d[(name, inst, wavelength)] = params['sourceId']  # doctest: +REMOTE_DATA
+
+        return(d)
```

Comment:
Why are there brackets around the dictionary?  Surely the dictionary is sufficient?

Reply:
Nope. We use the brackets to access the tuple

Reply:
Access the tuple? Maybe I am confused but that's the return statement, we don't need a bracket around the dictionary.

Reply:
Oh, I thought you were talking in the else statement
Yes, we can remove the brackets. It works in python 2 and 3

Reply:
Generally the lines people make the comments below are the ones we are referring to. Just something to be aware of. 

Also we only support Py3 as well.

Reply:
Sure I will keep that in mind